cpp/modmesh/toggle/RadixTree.hpp

```diff
@@ -179,5 +200,151 @@ class RadixTree
     key_type m_unique_id = 0;
 }; /* end class RadixTree */
 
+// The profiling result of the caller
+struct CallerProfile
+{
+    CallerProfile()
+        : total_time(0)
+    {
+    }
+
+    void start_stopwatch()
+    {
+        start_time = std::chrono::high_resolution_clock::now();
+        is_running = true;
+    }
+
+    void stop_stopwatch()
+    {
+        auto end_time = std::chrono::high_resolution_clock::now();
+        auto elapsed_time = std::chrono::duration_cast<std::chrono::nanoseconds>(end_time - start_time);
+        total_time += elapsed_time;
+        call_count++;
+    }
+
+    std::chrono::high_resolution_clock::time_point start_time;
+    std::function<void()> cancel_callback;
+    std::string caller_name;
+    std::chrono::nanoseconds total_time; /// use nanoseconds to have higher precision
+    int call_count = 0;
+    bool is_running = false;
+}; /* end struct CallerProfile */
+
+class CallProfilerTest; // for gtest
```

Comment:
Why don't you put it in gtest file?

Reply:
we cannot do that, otherwise it cannot build. If I put this line to gtests file, then we get:
```
  RadixTree.cpp
C:\Users\tiger\modmesh\cpp\modmesh/toggle/RadixTree.hpp(297,28): error C2433: 'modmesh::CallProfilerTest': 'friend' not permitted on data decla
rations [C:\Users\tiger\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
C:\Users\tiger\modmesh\cpp\modmesh/toggle/RadixTree.hpp(297,28): error C4430: missing type specifier - int assumed. Note: C++ does not support
default-int [C:\Users\tiger\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
```

Reply:
done